solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -91,6 +117,27 @@ public long fileLength(String name) throws IOException {
     }
   }
 
+  @Override
+  public long onDiskFileLength(String name) throws IOException {
+    Sizes ret = fileSizeMap.get(name);
+    SizeAccountingIndexOutput live;
+    if (ret != null) {
+      return ret.onDiskSize;
+    } else if ((live = liveOutputs.get(name)) != null) {
+      if (live.backing instanceof CompressingDirectory.SizeReportingIndexOutput) {
+        return ((CompressingDirectory.SizeReportingIndexOutput) live.backing).getBytesWritten();
+      }
+      // backing IndexOutput does not allow us to get onDiskSize
+      return 0;
+    } else if (in instanceof DirectoryFactory.OnDiskSizeDirectory) {
+      // fallback delegate to wrapped Directory
+      return ((DirectoryFactory.OnDiskSizeDirectory) in).onDiskFileLength(name);
+    } else {
+      // directory does not implement onDiskSize
+      return 0;
```

Comment:
here too I suspect that for consistency we should actually fallback to return simple `sizeOf()`.